scripts/modules/MsIndustryLinks/templates/data_source/DataSourceWorkflow.psm1

```diff
@@ -279,30 +279,12 @@ function New-LogicAppDataSourceWorkflow {
     # Set the custom connector properties if data source is CustomConnector
     $definition = Get-Content $PSScriptRoot/templates/data_source/$dataSourceType/logicapp_$dataSourceType.json | ConvertFrom-Json
     if ($isCustomConnector) {
-        $method = $WorkflowConfig.dataSource.properties.method.ToLower()
-        $definition.actions.Retrieve_data_using_custom_connector.inputs.host.connection.name = "@parameters('`$connections')['$apiName']['connectionId']"
-        $definition.actions.Retrieve_data_using_custom_connector.inputs.method = $method
-        $definition.actions.Retrieve_data_using_custom_connector.inputs.path = $WorkflowConfig.dataSource.properties.path
-
-        if ($null -ne $WorkflowConfig.dataSource.properties.queries) {
-            $definition.actions.Retrieve_data_using_custom_connector.inputs | Add-Member -MemberType NoteProperty -Name "queries" -Value $WorkflowConfig.dataSource.properties.queries
-        }
-        $bodyMethods = @("post", "put", "patch")
-        if ($null -ne $WorkflowConfig.dataSource.properties.body -and $method -in $bodyMethods) {
-            $definition.actions.Retrieve_data_using_custom_connector.inputs | Add-Member -MemberType NoteProperty -Name "body" -Value $WorkflowConfig.dataSource.properties.body
-        }
+        Set-LogicAppCustomConnectorConfiguration -WorkflowConfig $WorkflowConfig -Definition $definition | Out-Null
     }
 
     # Add data transform action if defined in the workflow configuration
     if ($null -ne $WorkflowConfig.dataTransform.type) {
-        $ingestDataSubflow = $definition.actions.Ingest_data_subflow
-        $dataTransformSubflow = Get-TransformDataSubflow -WorkflowName $WorkflowConfig.name -Text $ingestDataSubflow.inputs.body.payload -RunAfter $ingestDataSubflow.runAfter
-        $definition.actions | Add-Member -MemberType NoteProperty -Name "Transform_data_subflow" -Value $dataTransformSubflow
-
-        $definition.actions.Ingest_data_subflow.inputs.body.payload = "@body('Transform_data_subflow')"
-        $definition.actions.Ingest_data_subflow.runAfter = @{
-            Transform_data_subflow = @("Succeeded")
-        }
+        Set-LogicAppTransformConfiguration -WorkflowConfig $WorkflowConfig -Definition $definition | Out-Null
```

Comment:
Are you missing setting the variables to the new version or is powershell keeping track of the value of the variables?

Reply:
It's passed by reference